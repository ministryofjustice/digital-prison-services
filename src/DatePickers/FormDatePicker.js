import React, { Component } from 'react'
import PropTypes from 'prop-types'

import moment from 'moment'

import Datetime from 'react-datetime'
import 'react-datetime/css/react-datetime.css'
import LabelText from '@govuk-react/label-text'
import Label from '@govuk-react/label'
import ErrorText from '@govuk-react/error-text'
import Input from '@govuk-react/input'

class FormDatePicker extends Component {
  constructor(props) {
    super(props)
    this.renderInput = this.renderInput.bind(this)
  }

  renderInput(props) {
    const { label, meta, placeholder, input } = this.props
    return (
      <Label error={meta.error}>
        <LabelText> {label} </LabelText>
        {meta.touched && meta.error && <ErrorText>{meta.error}</ErrorText>}
        <Input
          {...props}
          name={input.name}
          error={meta.touched && meta.error}
          placeholder={placeholder}
          readOnly
          mb={6}
        />
      </Label>
    )
  }

  render() {
    const { shouldShowDay, input } = this.props

    return (
      <Datetime
        onChange={input.onChange}
        value={input.value}
        timeFormat={false}
        isValidDate={shouldShowDay}
        locale="en-GB"
        dateFormat="DD/MM/YYYY"
        closeOnSelect
        strictParsing
        renderInput={this.renderInput}
      />
    )
  }
}

FormDatePicker.propTypes = {
  shouldShowDay: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  label: PropTypes.string,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
  }),
  input: PropTypes.shape({
    onChange: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(moment)]),
  }).isRequired,
}

FormDatePicker.defaultProps = {
  placeholder: '',
  label: '',
  meta: {},
}

export default FormDatePicker
