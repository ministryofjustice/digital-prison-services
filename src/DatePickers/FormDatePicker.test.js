import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import FormDatePicker from './FormDatePicker'

describe('Form date picker', () => {
  beforeEach(() => {
    Date.now = jest.fn(() => new Date(Date.UTC(2017, 1, 2)).valueOf())
  })
  it('should render correctly', () => {
    const wrapper = renderer
      .create(
        <FormDatePicker
          shouldShowDay={() => {}}
          title="date"
          placeholder="Select"
          input={{ name: 'date', onChange: () => {}, value: '2017-01-01T:21:00:00Z' }}
          meta={{ touched: true, error: 'there was an error' }}
        />
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
