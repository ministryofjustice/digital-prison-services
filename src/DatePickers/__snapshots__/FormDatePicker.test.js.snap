// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form date picker should render correctly 1`] = `
<div
  className="rdt"
>
  <div>
    <label
      className="sc-bwzfXH cukpBy"
    >
      <span
        className="sc-bdVaJa jmnRGq"
      >
         
        
         
      </span>
      <span
        className="sc-htpNat gTYgEH"
      >
        there was an error
      </span>
      <input
        className="sc-bxivhb Qpyky form-control"
        name="date"
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select"
        readOnly={true}
        type="text"
        value="2017-01-01T:21:00:00Z"
      />
    </label>
  </div>
  <div
    className="rdtPicker"
  >
    <div
      className="rdtDays"
    >
      <table>
        <thead>
          <tr>
            <th
              className="rdtPrev"
              onClick={[Function]}
            >
              <span>
                ‹
              </span>
            </th>
            <th
              className="rdtSwitch"
              colSpan={5}
              data-value={1}
              onClick={[Function]}
            >
              February 2017
            </th>
            <th
              className="rdtNext"
              onClick={[Function]}
            >
              <span>
                ›
              </span>
            </th>
          </tr>
          <tr>
            <th
              className="dow"
            >
              Mo
            </th>
            <th
              className="dow"
            >
              Tu
            </th>
            <th
              className="dow"
            >
              We
            </th>
            <th
              className="dow"
            >
              Th
            </th>
            <th
              className="dow"
            >
              Fr
            </th>
            <th
              className="dow"
            >
              Sa
            </th>
            <th
              className="dow"
            >
              Su
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="rdtDay rdtOld rdtDisabled"
              data-month={0}
              data-value={30}
              data-year={2017}
            >
              30
            </td>
            <td
              className="rdtDay rdtOld rdtDisabled"
              data-month={0}
              data-value={31}
              data-year={2017}
            >
              31
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={1}
              data-year={2017}
            >
              1
            </td>
            <td
              className="rdtDay rdtToday rdtDisabled"
              data-month={1}
              data-value={2}
              data-year={2017}
            >
              2
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={3}
              data-year={2017}
            >
              3
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={4}
              data-year={2017}
            >
              4
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={5}
              data-year={2017}
            >
              5
            </td>
          </tr>
          <tr>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={6}
              data-year={2017}
            >
              6
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={7}
              data-year={2017}
            >
              7
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={8}
              data-year={2017}
            >
              8
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={9}
              data-year={2017}
            >
              9
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={10}
              data-year={2017}
            >
              10
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={11}
              data-year={2017}
            >
              11
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={12}
              data-year={2017}
            >
              12
            </td>
          </tr>
          <tr>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={13}
              data-year={2017}
            >
              13
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={14}
              data-year={2017}
            >
              14
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={15}
              data-year={2017}
            >
              15
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={16}
              data-year={2017}
            >
              16
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={17}
              data-year={2017}
            >
              17
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={18}
              data-year={2017}
            >
              18
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={19}
              data-year={2017}
            >
              19
            </td>
          </tr>
          <tr>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={20}
              data-year={2017}
            >
              20
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={21}
              data-year={2017}
            >
              21
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={22}
              data-year={2017}
            >
              22
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={23}
              data-year={2017}
            >
              23
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={24}
              data-year={2017}
            >
              24
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={25}
              data-year={2017}
            >
              25
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={26}
              data-year={2017}
            >
              26
            </td>
          </tr>
          <tr>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={27}
              data-year={2017}
            >
              27
            </td>
            <td
              className="rdtDay rdtDisabled"
              data-month={1}
              data-value={28}
              data-year={2017}
            >
              28
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={1}
              data-year={2017}
            >
              1
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={2}
              data-year={2017}
            >
              2
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={3}
              data-year={2017}
            >
              3
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={4}
              data-year={2017}
            >
              4
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={5}
              data-year={2017}
            >
              5
            </td>
          </tr>
          <tr>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={6}
              data-year={2017}
            >
              6
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={7}
              data-year={2017}
            >
              7
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={8}
              data-year={2017}
            >
              8
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={9}
              data-year={2017}
            >
              9
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={10}
              data-year={2017}
            >
              10
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={11}
              data-year={2017}
            >
              11
            </td>
            <td
              className="rdtDay rdtNew rdtDisabled"
              data-month={2}
              data-value={12}
              data-year={2017}
            >
              12
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
